

# Fall 2022 STA130: An Introduction to Statistical Reasoning and Data Science

> *Welcome to your Fall 2022 semester and STA130!* **STA130 Lectures** are Monday's 10:10AM/2:10PM ET for L0101/L0201, and **STA130 Tutorials** are Friday's 10:10AM/2:10PM ET for L0101/L0201. The Fall 2022 term begins September 8, but we won't meet in person for a tutorial on the first Friday, September 9th; instead, you are expected to use your two-hour tutorial block to review the course website and get a head start on familiarizing yourself with R based on your review of the course website.  See you in Lecture on Monday, September 12th!

---

Prior to Winter 2020 this was an **In-Person** course (e.g., [F18](https://ntaback.github.io/UofT_STA130/Fall2018/fall2018_course_content.html), 
[F19](https://www.statistics.utoronto.ca/sites/www.statistics.utoronto.ca/files/STA130_syllabus_F19.pdf), 
[W20](https://www.statistics.utoronto.ca/sites/www.statistics.utoronto.ca/files/STA130H1S.pdf)).
From Fall 2020 to Winter 2022 it was transitioned to an **Online** course (e.g., [W22](https://q.utoronto.ca/courses/253019)).
The current course transitions these iterative course developments back into the previous **In-Person** format. Specifically,
the Winter 2022 Online [.ppt and pre-recording material](https://q.utoronto.ca/courses/253019) is integrated into leveraged and reworked
[Rstudio Rmd](https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.dropbox.com%2Fl%2Fscl%2FAACmrT8hzMyabinD1Ioc4_CWCLbgwhZx4lU&data=05%7C01%7Cscott.schwartz%40utoronto.ca%7C07bd03b4049f4302640108da3e67cd9b%7C78aac2262f034b4d9037b46d56c55210%7C0%7C0%7C637890913830493523%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C2000%7C%7C%7C&sdata=7uY5HYCnQ9Rqp4U5KOSh9UigGyzhCfymDSehiQSC6xA%3D&reserved=0) and [beamer pdf](https://ntaback.github.io/UofT_STA130/Fall2018/fall2018_course_content.html) materials of the previous **In-Person** courses. 
At the highest level, the course activities are meant to develop and practice the two steps the statistical and data science workflow:

1. Extract meaning from data through coding and analysis
2. Communicate findings through writing and speaking

More specifically, STA130 has the following **5 Learning Objectives**:

1. Implement the computational steps involved in the management and statistical analysis of data using R.
2. Carry out a variety of statistical analyses in R and interpret the results of the analyses.
3. Clearly communicate the results of statistical analyses to technical and non-technical audiences.
4. Identify appropriate uses of statistical methods to answer questions, including their strengths and weaknesses.
5. Describe how statistical methods can be used to learn from data, including methods for description, explanation, and prediction.

## Schedule

The course (on average across students) should take 10 hours a week with 7 hours allocated for

| Section     |             | Monday     | Thursday       | Friday     |
|-------------|-------------|------------|----------------|------------|
|  L0101      | 10:10 AM ET | Lec (2hrs) |                | In-Person Tut work (2hrs) |
|  L0201      | 2:10 PM ET  | Lec (2hrs) |                | In-Person Tut work (2hrs) | 
| L0101+L0201 | 5:00 PM ET  |            | R (2hr) HW Due |            |
| L0101+L0201 | 10:00 PM ET |            |                | Tut work (+1hr) Due |

and 3 hours left available for study and review, office hours and piazza discussion boards, and eventual team project work.

## Grading

| Components             | Notes                | Absence Policy            | Accomodation Requests | 
|------------------------|----------------------|---------------------------|-----------------------|
| 7% R Homework          | 9 assignments        | Best 7/9 are Scored       | See below             |
| 14% In-Person Tut work | 9 activities         | Best 7/9 are Scored       | See below             |
| 20% Course project     |                      | Due Dec 8th               | Emergency Absenses ONLY  |
| 20% Midterm            |                      | Reweights to Final if Missed | Automatic          |
| 34% Final exam         |                      | Excused Absence Rescheduling | Handled through FAS |
| 5% Participation       | Mentorship + Surveys | Due Dec 8th               | No Extensions Available |
| N/A Lectures           | Not Recorded         | N/A                       | Lecture `.Rmd`+Slides Available |
| N/A Study Material     | Not Graded           | Optional Material         | N/A                   |


#### Best 7/9 policy

For students having a "normal" experience acclimating to the UofT context, 
this should provide a comfortable buffer against minor unexpected events or schedule management growing pains.
For example, if you're not feeling well and think it's best to stay home on a given day,
this policy allows you to confidently make that choice (and help build increasing confidence in our in-person environment)! 
Contingencies involving **three absences** due to health related issues can be initiated through your [College Registrar](https://www.artsci.utoronto.ca/current/academic-advising-and-support/college-registrars-offices) and can be accomodated upon the recommendation of the Registrar with the following limitations:

- Accomodations involving **three absences** due to health related issues will be considered provided that notice of the advent of special circumstances is made to both (a) the [College Registrar](https://www.artsci.utoronto.ca/current/academic-advising-and-support/college-registrars-offices) and (b) sta130@utorono.ca within one week of return to University activities. 
- Accomodations beyond the "Best 7/9 policy" involving one or two health related absences will not be considered.
- Accomodations beyond the "Best 7/9 policy" involving time management challenges will not be considered.


## Help

| One on One Help | General Questions | 
|-|-|
| TA Online Office Hours | Online Discussion Board |


| Accomodation | Contingency Request Recipient(s) | Notes |
|-|-|-|
| Best 7/9 policy | [College Registrar](https://www.artsci.utoronto.ca/current/academic-advising-and-support/college-registrars-offices) + sta130@utorono.ca | Notice must be provided within one week of return to UofT activities |
| Final Exam | [FAS Exam Deferrel Petition](https://www.artsci.utoronto.ca/current/faculty-registrar/petitions/deferred-exams) | The Faculty of Arts and Sciences administers the final exam |
| Miderm Exam | Final Reweighting | This is automatic so no request is required | 
| Course Porject | sta130@utorono.ca | Extensions will be granted only for Emergency Absenses ONLY |

## Calendar

Fall 2022 semester begins Thursday, September 8th.  Friday, September 9th there is no tutorial. 
The first day of class and the first day of tutorial occur during the week of September 12th.

| Week of | Topic                                     | Monday Lec     | Thursday 5PM | Friday Tut | Friday 10PM |
|---------|-------------------------------------------|----------------|--------------|-------------------|------------|
| Sep 12  | Jupyterhub, Rstudio, R Basics             | 10:10AM/2:10PM | HW1 Due      | 10:10AM/2:10PM    | Tut 1 Due  |
| Sep 19  | Distributions and Statistics              | 10:10AM/2:10PM | HW2 Due      | 10:10AM/2:10PM    | Tut 2 Due  |
| Sep 26  | Data Wrangling with Tidy                  | 10:10AM/2:10PM | HW3 Due      | 10:10AM/2:10PM    | Tut 3 Due  |
| Oct 3   | One and Two Sample Inference              | 10:10AM/2:10PM | HW4 Due      | 10:10AM/2:10PM    | Tut 4 Due  |
| Oct 10  | *THANKSGIVING DAY: NO CLASS, RESCHEDULED  |                |              | 10:10AM/2:10PM    | **Projet Introduction** |
| Oct 17  | Sampling Dists and Bootstrap CIs             | 10:10AM/2:10PM | HW5 Due      | 10:10AM/2:10PM    | Tut 5 Due  |
| Oct 24  | **Midterm Review**                        | 10:10AM/2:10PM |              | 10:10AM/2:10PM    | **Midterm Exam**   |
| Oct 31  | Linear Regression I                       | 10:10AM/2:10PM | HW6 Due      | 10:10AM/2:10PM    | Tut 6 Due  |
| Nov 7   | READING WEEK                              |                |              |                   |            |
| Nov 14  | Linear Regression II                      | 10:10AM/2:10PM | HW7 Due      | 10:10AM/2:10PM    | Tut 7 Due  |
| Nov 21  | Classification Trees                      | 10:10AM/2:10PM | HW8 Due      | 10:10AM/2:10PM    | Tut 8 Due  |
| Nov 28  | Study Design, Confounding, Ethics         | 10:10AM/2:10PM | HW9 Due      | 10:10AM/2:10PM    | Tut 9 Due  |
| Dec 5   | **Final Review**                          | 10:10AM/2:10PM |              |                   |            |
|         | *Oct 10 -> Dec 8th Reschedule             |                | **Thursday** |                   |            |
|         |                                           |                | **Dec 8th**  |                   |            |
|         |                                           |                | **10:10AM/2:10PM**        |      |            |
|         |                                           |                | **Project (DUE)**  |                   |            |
|         |                                           |                | **Presentations** |              |            |


## Course project

The course project will be done in consultation with our "project collaborator" [Dr. Heman Shakeri](https://datascience.virginia.edu/people/heman-shakeri) of the [University of Virginia's](https://www.virginia.edu/) (UVA) [Data Science Institute](https://datascience.virginia.edu/) (DSI). The DSI is a recently created [School](https://datascience.virginia.edu/pages/uva-plans-new-school-data-science) (the 12th) of this [premier](https://www.usnews.com/best-colleges/uva-6968/overall-rankings) US university originally founded by [Thomas Jefferson](https://www.virginia.edu/aboutuva) the [principal author](https://en.wikipedia.org/wiki/Thomas_Jefferson) of the US [Declaration of Independence](https://en.wikipedia.org/wiki/United_States_Declaration_of_Independence).

Dr. Shakeri's research, in conjuction with the [Department of Biomedical Engineering](https://engineering.virginia.edu/departments/biomedical-engineering) and 
[Systems Biology and Biomedical Data Sciences](https://engineering.virginia.edu/departments/biomedical-engineering/research/systems-biology-and-biomedical-data-sciences),
is motivated by the experience with cancer of a close family friend. So, Heman (Dr. Shakeri) wants to use "data-driven identification and control of high-dimensional dynamical systems" to detect deviations away from normal cellular function and intervene to interrupt the pregression of cancer before it can establish a deleterious cellular homeostasis in order to  give family's more time with their loved ones and close friends.

The data we will work with is based on advances in the fields of [Flow Cytometry](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5939936/) for single cell analysis and [Mass Spectrometry](https://www.broadinstitute.org/technology-areas/what-mass-spectrometry) for measurement of cellular proteomic processes (the [phenotypical process endpoint](https://en.wikipedia.org/wiki/Central_dogma_of_molecular_biology) of cellular function and behavior). Based on these technologies, the multivariate landscape of proteomic activity can be measured for a single cell in any experiemental condition for any cell type (e.g., cancerous and benign cellular lines) at scale. By understanding typical cellular homeostatis of healthy and deliterious cells, and observing the phenotypical transformation of cellular proteomic homeostatsis over time in response to different treatments, it is hoped that we will eventually understand how to direct deleterious cellular states to transition into non-deleterious states. I.e., "data-driven identification and control of high-dimensional dynamical systems".

The data observations simultaneously measure 17 so-called [AP-1 transcription factors](https://en.wikipedia.org/wiki/AP-1_transcription_factor) over thousands of cells in a given experiemental condition. By repeatedly observing this protein complex (as is done in this data), the correlation between the different proteins can be observed. Further, the evolution of these dependencies can be observed over time as a response to different interventions. And even further, the emergence of downstream cellular phenotypes in response to changes of the state of the AP-1 system can also be observed and can be characterized through 4 other "phenotype" proteins (whose measurements are also available in this data set). Thus, though our course project, we seek to understand the inter-dependence between the AP-1 proteins, and their driving relationship with downstream cellular phenotypes, which might eventually suggest how we can intervene along this pathway to induce transformation away from deletarious cellular states. 

## Outline

### Week 1: Jupyterhub and Rstudio and R Basics

Week 1 is concerned with introducing students to [R](https://www.r-project.org/) (and packages like [tidyverse](https://www.tidyverse.org/)) and [Rstudio](https://www.rstudio.com/) using [UofT's Jupyterhub](https://jupyter.utoronto.ca). Our primary reference resources in this task are

- the [R for Data Science](https://r4ds.had.co.nz/workflow-basics.html) textbook by Hadley Wickham & Garret Grolemund
  - or Garret Grolemund's previous incarnation [Hands-On Programming with R](https://d1b10bmlvqabco.cloudfront.net/attach/ighbo26t3ua52t/igp9099yy4v10/igz7vp4w5su9/OReilly_HandsOn_Programming_with_R_2014.pdf)
- the [DoSS Toolkit](https://dosstoolkit.com/) created by seasoned STA130 Profs. Alexander and Caetano et al.
  - specifically the [Rstudio](https://dosstoolkit.com/#hello-world), [errors](https://dosstoolkit.com/#operating-in-an-error-prone-world),
and [packages](https://dosstoolkit.com/#holding-the-chaos-at-bay) [tutorials](https://dosstoolkit.com/#if-you-have-never-used-r-before)
- R+Rstudio [primers](https://rstudio.cloud/learn/primers/1.2) and [Rmarkdown](https://raw.githubusercontent.com/rstudio/cheatsheets/main/rmarkdown.pdf) and [Rstudio](https://raw.githubusercontent.com/rstudio/cheatsheets/main/rstudio-ide.pdf) cheatsheets
- and [free online introductions to R](https://www.datacamp.com/courses/free-introduction-to-r) from learning websites like [datacamp](https://www.datacamp.com/).

> The [UofT Jupyterhub](https://jupyter.utoronto.ca) is a phenomenal resource; however, it is subject to service outages from time to time (which have in the past coincided with assignment due dates), and it can take a long time to load when there's a lot of simultaneous user demand (if a lot of students in our or another class log in at once). When you cannot use [UofT Jupyterhub](https://jupyter.utoronto.ca) you must use your own [local Rstudio instance](https://www.rstudio.com/).

> An extremely valuable skill in the context of coding for statistics and Data Science is troubleshooting and figuring things out.  Resources like the [R for Data Science](https://r4ds.had.co.nz/) textbook and the the [DoSS Toolkit](https://dosstoolkit.com/) are excellent recources to learn things in a systematic, structured, and organized manner; however, google, stack exchange/overflow, and coding blog posts can be an invaluable resource for finding quick solutions for coding bugs and suggestions for how to complete a desired analyses. Hopefully through this class you will take the opportunity to build your self-sufficiency and coding-resiliance.

#### Week 1 Course Material

| [Slides](https://github.com/pointOfive/STA130_Week1_Slides) [[Jupyterhub](https://jupyter.utoronto.ca/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2FpointOfive%2FSTA130_Week1_Slides&urlpath=rstudio%2F&branch=main)], [Demo 1](https://github.com/pointOfive/STA130_Week1_Demo1) [[Jupyterhub](https://jupyter.utoronto.ca/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2FpointOfive%2FSTA130_Week1_Demo1&urlpath=rstudio%2F&branch=main)], [Demo 2](https://github.com/pointOfive/STA130_Week1_Demo2) [[Jupyterhub](https://jupyter.utoronto.ca/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2FpointOfive%2FSTA130_Week1_Demo2&urlpath=rstudio%2F&branch=main)] | [Questions](https://pollev.com/sta) [[Round 1](https://PollEv.com/surveys/Sn8Y5ZdhDdocnPBg5lWU3/respond), [Round 2](https://PollEv.com/surveys/ZajOSOpWDmepur3hTlXKV/respond), [Round 3](https://PollEv.com/surveys/ytOu9cYQoL6Oyon1dmUWr/respond), [Round 4](https://PollEv.com/surveys/5gpW9MIWgzTuE81nEJDaH/respond), [Round 5](https://PollEv.com/surveys/k46QXuZy1j1a9kuXy4eY0/respond)] |
|-|-|
| [Homework Assignment](https://github.com/SamanthaJoCaetano/STA130-W22-ProblemSet1) and [Practice Quiz](https://q.utoronto.ca/courses/253019/quizzes/236277/edit) | |


### Module 2: Distributions and Statistics

### Module 3: Data Wrangling with Tidy

### Module 4: Statistical Inference for a Single Sample Proportion

### Module 5: Permutation Tests for Two Groups

### Midterm Review

### Module 6: Sampling Distributions and Bootstrap Confidence Intervals

### Module 7: Linear Regression I

### Module 8: Linear Regression II

### Module 9: Classification Trees

### Module 10: Study Design, Confounding, and Ethics

### Final Review
